src/main/java/com/artipie/auth/AuthFromYaml.java

```diff
@@ -66,7 +59,8 @@ public AuthFromYaml(final YamlMapping cred) {
         Optional<String> res = Optional.empty();
         if (users != null && users.yamlMapping(user) != null) {
             final String stored = users.yamlMapping(user).string("pass");
-            if (stored != null && check(stored, pass)) {
+            final String type = users.yamlMapping(user).string("type");
+            if (stored != null && type != null && check(stored, type, pass)) {
```

Comment:
@paulodamaso I would move null checks to `check` method to make it more readable or add another method `validParams` that would check against null. wdyt?

Reply:
@Vatavuk I don't think it's a good idea, here's why:
- `validParams` wouldn't mean nothing here, since `stored` and `type` are not login parameters, they are data stored into Yaml file instead: if they are null, means that we don't have these field for the current user;
- `check` method just checks if the two values match plainly or digested, so I don't think that we should move these tests inside this method

Reply:
@paulodamaso ok